protocol/binary/writer.go

```diff
@@ -59,63 +58,29 @@ type Writer struct {
 //
 // This Writer must be returned back using ReturnWriter.
 func BorrowWriter(w io.Writer) *Writer {
+	streamWriter := streamWriterPool.Get().(*StreamWriter)
+	streamWriter.writer = w
```

Comment:
Should this just BorrowStreamWriter instead of accessing the pool directly?

```suggestion
	streamWriter := BorrowStreamWriter(w)
```